torchrl/objectives/a2c.py

```diff
@@ -445,7 +475,26 @@ def loss_critic(self, tensordict: TensorDictBase) -> torch.Tensor:
                 f"TDLambdaEstimate and TDEstimate all return a 'value_target' entry that "
                 f"can be used for the value loss."
             )
-        return self.critic_coef * loss_value
+        clip_fraction = None
+        if self.clip_value:
+            clip_value = self.clip_value.to(state_value.device)
+            state_value_clipped = old_state_value + (
+                state_value - old_state_value
+            ).clamp(-clip_value, clip_value)
+            loss_value_clipped = distance_loss(
+                target_return,
+                state_value_clipped,
+                loss_function=self.loss_critic_type,
+            )
+            # Chose the most pessimistic value prediction between clipped and non-clipped
+            loss_value = torch.max(loss_value, loss_value_clipped)
+            clip_fraction = (
```

Comment:
I don't get what clip_fraction represents

One thing I like to look at is what proportion of data is being clipped

Also we should be careful about this since it doesn't come for free. The overhead introduced by logging this metric could impact performance...

Reply:
Yes, as I understand it, it is a measure of how much the old model that collected the data differs from the current model. If the fraction values stay close to 1.0 the model is changing slowly.  If the ratio consistently deviates from 1.0 and approaches or hits the clipping value, it indicates that the new model is substantially different from the old policy, indicating a faster rate of policy change.